pkg/types/conversion/installconfig.go

```diff
@@ -168,6 +168,27 @@ func convertOpenStack(config *types.InstallConfig) error {
 		config.Platform.OpenStack.DefaultMachinePlatform.FlavorName = config.Platform.OpenStack.DeprecatedFlavorName
 	}
 
+	// NOTE: THIS DOES NOT WORK YET (see tests)
+	//
+	// type has been deprecated in favor of types in the machinePools.
+	if config.ControlPlane != nil && config.ControlPlane.Platform.OpenStack.RootVolume != nil && config.ControlPlane.Platform.OpenStack.RootVolume.DeprecatedType != "" {
+		if config.ControlPlane.Platform.OpenStack.RootVolume.Types != nil && len(config.ControlPlane.Platform.OpenStack.RootVolume.Types) > 0 {
```

Comment:
nit: a nil slice has length zero in Go.

```suggestion
		if len(config.ControlPlane.Platform.OpenStack.RootVolume.Types) > 0 {
```

Reply:
you already check that rootVolume is not nil in this block.